cEP-0033.md

```diff
@@ -0,0 +1,760 @@
+
+		Returns the NlSection object
+		"""
+
+		return NlSection.from_values(start=start, end=end, language=language,
+					     index=index)
+
+```
+
+### Making temporary files
+
+This sections deals with how the segregated section (nl_sections) from the 
+parser are combined to form different temporary files on which linting
+will be done. 
+
+`NlFileHandler` is responsible to create the temporary files. These temporary 
+files are stored in the memory.
+
+1. The nl_sections are passed to NlFileHandler.
+
+2. NlFileHander then creates a `nl_file_dict`, where the key is the name of 
+the temporary file and the contents are it's value. It is similar to the
+`file_dict` that is created by coala in `instantiate_process()` during the
+execution of the section.
+
+3. It is important to keep in mind, that the temporary segregated file are not
+actually present in the folder. In the normal flow of coala,during the execution 
+of the coala sections, coala will try to find the files by the filenames 
+mentioned in the coala section. And then create a `file_dict`.
+That cannot happen in our case. So we will explicitly replace the `file_dict` 
+with `nl_file_dict`.   
+
+### NlFileHandler.py Prototype
+
+```python
+def get_nl_file_dict(nl_file_info, nl_sections):
+	'''
+	Returns a nl_file_dict, where the key is the name of the temporary file
+	and the value is the contents of that file.
+
+	:param nl_file_info:	The information extracted from the arguments.
+				This is generated by the NlInfoExtractor.
+	:param nl_sections:	The segregated sections of the original file.
+
+	>>> nl_file_info
+	{
+		"file_name": html_php_file.html,
+		"absolute_file_path": /home/test/html_php_file.html,
+		"languages": [html, php]
+		"bears": [HTMLBear, PHPBear]
+		"language_bear_dict": language_bear_dict
+		"arg_dict": arg_dict
+	}
+
+	>>> nl_sections
+	(	<NlSection object(index=1, language=html, orig_start=1, orig_end=3)>,
+		<NlSection object(index=2, language=php, orig_start=4, orig_end=6)>,
+		<NlSection object(index=3, language=html, orig_start=7, orig_end=7)>
+	)
+
+	>>> file_contents = get_file(nl_file_info['file_name'])
+
+	>>> file_contents = \
+			"""
+			<!DOCTYPE html>
+ 			<head>
+			<title>Hello world as text</title>
+			<?php
+				echo "<p>Hello world</p>";
+			?>
+			</html>
+			"""
+
+	>>> nl_file_dict = get_nl_file_dict(nl_file_info, nl_sections)
+	>>> nl_file_dict
+	{
+		"temp_html_file":("<!DOCTYPE html>\n
+				   <head>\n
+				   <title>Hello world as text</title>\n
+				   \n
+				   \n		
+				   \n
+				   </html>"),
+
+		"temp_php_file":('\n
+				  \n
+				  \n
+				  \t\t<?php\n
+				  \t\t\techo "<p>Hello world</p>";\n
+				  \t?>')
+
+
+	}
+
+	'''
+
+def get_file_contents(file_name):
+	""" Returns the contents of the file"""
+
+def assemble(nl_diff_dict, nl_sections, nl_info_dict):
+	"""
+	Assembles the temporary files. 
+
+	The sections are extracted by their increasing order of their index and 
+	then written directly to the original file.
```

Comment:
Not sure, but probably coala has some functions that make a backup of the original file as `*.orig`  
Just a note that check them out once and if there are any reusable parts then try to use them

Reply:
afaik, There is no standalone function which make the `.*orig` file. Rather there is a small snippet of code in `ApplyPatchAction.apply()`, which makes  a `.orig` file and write the content of the diff into the file.

But yes, I will be making use of the same snippet as reference to write the linted file , by making the `.*orig` file.

Reply:
Cool
Mentioned it once here since couldn't infer from the docstring nor the code if you were going to take a backup or something :sweat_smile: